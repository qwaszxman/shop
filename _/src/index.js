
// import React from 'react';
// import ReactDOM from 'react-dom';
// import './index.css';
// import App2 from './Components/App';
// import App from './Components/App';
import * as serviceWorker from './serviceWorker';



import 'babel-polyfill'
import React from 'react';
import ReactDOM from 'react-dom';

import App from './asample/components/App';
import Root from './asample/Root';

import './index.scss';

ReactDOM.render(
    <Root>
        <App />
    </Root>,
    document.getElementById('root')
);



// ReactDOM.render(<App />, document.getElementById('root'));

// If you want your app to work offline and load faster, you can change
// unregister() to register() below. Note this comes with some pitfalls.
// Learn more about service workers: https://bit.ly/CRA-PWA
serviceWorker.unregister();
