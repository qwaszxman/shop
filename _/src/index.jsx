import 'babel-polyfill';
import React from 'react';
import ReactDOM from 'react-dom';
import * as serviceWorker from './serviceWorker';

import App from './Components/App';
import Root from './Root';
import { StoreProvider } from "./Store/store";

import './index.scss';

// ReactDOM.render(
//   <Root>
//     <App />
//   </Root>,
//   document.getElementById('root')
// );

ReactDOM.render(
  <StoreProvider>
    <App />
  </StoreProvider>,
  document.getElementById('root')
);

serviceWorker.unregister();
