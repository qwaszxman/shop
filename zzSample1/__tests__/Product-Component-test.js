/*
 * Product tests for JEST
 *
 * Copyright © Oleg Nosov 2016-Present
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE.txt file in the root directory of this source tree.
 */

import React from 'react';
import renderer from 'react-test-renderer';
import IntlMessageFormat from 'intl-messageformat';
import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Product from '../src/components/Product/Product';
import { generateProductKey } from '../src/helpers';

Enzyme.configure({ adapter: new Adapter(), });

const testProductLocalization = {
  color: 'Color',
  iPadCase: 'iPad case',
  red: 'Red',
  green: 'Green',
  price: 'Price: {localizedCurrency}{price}',
  GBP: '£',
  quantityLabel: 'Quantity:',
  propertyLabel: '{localizedName}:',
  addToCart: 'Add to my cart :)',
};

const getLocalization = (id, params = {}) =>
  new IntlMessageFormat(testProductLocalization[id], 'en').format(params);

const createProduct = ({ cartState, props, }, renderFunc = mount) =>
  renderFunc(<Product
    {...props}
    onAddProduct={
      (
          key,
          product,
      ) => {
          cartState[key] = product;
      }
    }
    checkoutButton={<a />}
    getLocalization={
        getLocalization
      }
    currency="GBP"
    generateProductKey={generateProductKey}
  />);


describe('Product', () => {
  const iPadCaseProps = {
    name: 'ipadCase',
    id: 'ipad-case',
    path: '/shop/ipad-case/',
    properties: {
      color: [ 'red', 'green', ],
    },
    propertiesToShowInCart: [ 'color', ],
    prices: { GBP: 70, },
    currency: 'GBP',
    imageSrc: '1-483x321.jpeg',
  };

  it('adds products to cart', () => {
    // Create empty cart
    const cartState = {};

    // Current product is ipad case (see props above)
    const renderedProduct = createProduct({ cartState, props: iPadCaseProps, });

    const productKey = generateProductKey(
      iPadCaseProps.id,
      { color: 'red', },
    );

    const simulateAddProductEvent =
      () => renderedProduct.find('form').simulate('submit');

    simulateAddProductEvent();

    // Our product is in cart already
    expect(cartState[productKey].quantity).toBe(1);

    const quantityInput = renderedProduct.find('input');

    // Try to change quantity to -1
    quantityInput.instance().value = -1;

    renderedProduct.find('input').simulate('change');

    expect(renderedProduct.update().state().quantity).toBe(1);

    const colorSelect =
      renderedProduct
        .find('select');

    // Now add green case in our cart
    colorSelect.instance().value = 'green';

    renderedProduct
      .find('select')
      .simulate('change');

    simulateAddProductEvent();

    // Have two products in cart
    expect(Object.keys(cartState).length).toBe(2);
  });

  it('takes snapshot', () => {
    const renderedProduct = createProduct(
      { props: iPadCaseProps, },
      renderer.create,
    );
    expect(renderedProduct.toJSON()).toMatchSnapshot();
  });
});
