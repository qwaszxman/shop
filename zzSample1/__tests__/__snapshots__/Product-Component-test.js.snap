// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product takes snapshot 1`] = `
<div>
  <p>
    Price: Â£70
  </p>
  <form
    className="my-1"
    onSubmit={[Function]}
  >
    <div
      className="form-group row"
    >
      <label
        className="col-xs-3 col-sm-3 col-md-3 col-lg-3 col-form-label"
        htmlFor="color"
      >
        Color:
      </label>
      <div
        className="col-xs-9 col-sm-9 col-md-9 col-lg-9"
      >
        <select
          className="form-control"
          onChange={[Function]}
          value="red"
        >
          <option
            value="red"
          >
            Red
          </option>
          <option
            value="green"
          >
            Green
          </option>
        </select>
      </div>
    </div>
    <div
      className="form-group row"
    >
      <label
        className="col-xs-3 col-sm-3 col-md-3 col-lg-3 col-form-label"
        htmlFor="product-quantity"
      >
        Quantity:
      </label>
      <div
        className="col-xs-9 col-sm-9 col-md-9 col-lg-9"
      >
        <input
          className="form-control"
          onChange={[Function]}
          type="number"
          value={1}
        />
      </div>
    </div>
    <div
      className="form-group row"
    >
      <div
        className="col-xs-12 col-sm-12 col-md-8 col-lg-6 col-xl-6 m-auto"
      >
        <button
          className="btn btn-success btn-block"
          disabled={false}
          role="button"
          type="submit"
        >
          <i
            className="fa fa-cart-plus mx-1"
          />
          Add to my cart :)
        </button>
        <a />
      </div>
    </div>
  </form>
</div>
`;
